Index: src/path.c
===================================================================
--- src/path.c	(revision 32661)
+++ src/path.c	(working copy)
@@ -1084,6 +1084,25 @@ int git_path_direach(
 
 #if defined(GIT_WIN32) && !defined(__MINGW32__)
 
+static FINDEX_INFO_LEVELS find_first_type()
+{
+  static int lvl = 0;
+
+  if (!lvl)
+    {
+      OSVERSIONINFO osvi = { 0 };
+      osvi.dwOSVersionInfoSize = sizeof(osvi);
+
+      GetVersionEx(&osvi);
+
+      lvl = (osvi.dwMajorVersion <= 5) ? 1 : 2;
+    }
+
+  return (lvl == 1)
+        ? 0   /* FindExInfoStandard: Include short names */
+        : 1;  /* FindExInfoBasic: No short names */
+}
+
 /* Using _FIND_FIRST_EX_LARGE_FETCH may increase performance in Windows 7
  * and better.  Prior versions will ignore this.
  */
@@ -1122,7 +1141,7 @@ int git_path_diriter_init(
 
 	diriter->handle = FindFirstFileExW(
 		path_filter,
-		FindExInfoBasic,
+		find_first_type(),
 		&diriter->current,
 		FindExSearchNameMatch,
 		NULL,
